CUSTOM_IMAGE.md

```diff
@@ -0,0 +1,76 @@
+Building a Custom Spark Image
+=============================
+
+Overview
+--------
+To build a custom Spark image, with a custom version of Spark or other libraries, 
+such as Hadoop, Python, or R, follow the steps below. This build does the following:
+1. Builds a Docker image containing Spark and other libraries.
+1. Builds a DC/OS package (also called a "universe") based on this Docker image.
+1. Uploads this package to Amazon S3.
+
+Prerequisites
+-------------
+* [AWS Command Line Interface](https://aws.amazon.com/cli/) (CLI)
+* The [jq](https://github.com/stedolan/jq) JSON parser
+
+Custom Spark Version, Hadoop Version
+------------------------------------
+Edit `manifest.json`, and set `spark_uri` to point to a custom Spark build. This custom build may 
+contain a custom version of Spark or Hadoop.
+To build a custom version of Spark, run
+`./dev/make-distribution.sh` in your Spark project directory, according to the instructions
+[here](http://spark.apache.org/docs/latest/building-spark.html#building-a-runnable-distribution). 
+Make sure to set the sparkr flag if you will be using SparkR.
+
+Custom Python, R Versions
+-------------------------
+Edit `docker/Dockerfile` to install custom versions of Python or R in the Docker image.
```

Comment:
I think it would be simpler to have them extend our docker image, rather than modifying it: 

```
FROM mesosphere/spark:1.0.4-2.0.1
<custom commands>
```

Modifying the Dockerfile breaks encapsulation, and increases the risk of them mucking up the image.